doc/api/module.md

```diff
@@ -111,6 +111,15 @@ changes:
 Register a module that exports [hooks][] that customize Node.js module
 resolution and loading behavior. See [Customization hooks][].
 
+> **Warning** If setting up a `MessageChannel` to communicate with a loader
+> within the Module Worker, beware that this can lead to a deadlock. For
+> example, you have 2 modules, A and B. "A" is registered first and sets up a
+> message channel,which it uses in its `resolve` hook. "B" uses `register` to
+> register its own loader. Resolving "B"'s own loader will go through "A"'s
+> `resolve`, which will try to communicate with the Module Worker. The Module
```

Comment:
```suggestion
> `resolve`, which will try to communicate with the thread that the hooks are running on. The Module
```

No one knows what Module Worker means. I'm not even sure what it means.

Reply:
Out of curiosity, who is "no one"?